sourcegraph/cody
# Edit: Inline Diff

## Description

This PR adds:
- An inline diff view for all Edit commands (except document command, due to [this](https://github.com/sourcegraph/cody/issues/4628))
- An improved animation showing the progress of the LLM before it is applied
- Improved formatting logic that implements indentation matching and some basic multi-language format support (currently just to remove semi colons from responses if the file doesn't use them in JS/TS)

https://github.com/sourcegraph/cody/assets/9516420/d4b15c9e-a797-4686-ba17-8a2d3dd91718



## Test plan

1. Make Edits:
- Edit
- Generate
- Fix
- Document
- Test

2. Check that decorations are shown correctly

3. Check the decorations are removed correctly on task complete (save/edit in task range/undo task)
